src/explore-education-statistics-common/src/components/ContentHtml.tsx

```diff
@@ -1,31 +1,104 @@
 import sanitizeHtml, { SanitizeHtmlOptions } from '@common/utils/sanitizeHtml';
-import htmlParser from 'html-react-parser';
+import parseHtmlString, { DOMNode, domToReact } from 'html-react-parser';
+import { Element } from 'domhandler/lib/node';
 import classNames from 'classnames';
-import React, { useMemo } from 'react';
+import React, { useMemo, useState } from 'react';
+import { GlossaryEntry } from '@common/services/types/glossary';
+import Modal from '@common/components/Modal';
 
 export interface ContentHtmlProps {
   className?: string;
   html: string;
   sanitizeOptions?: SanitizeHtmlOptions;
   testId?: string;
+  getGlossaryEntry?: (slug: string) => Promise<GlossaryEntry>;
 }
 
 const ContentHtml = ({
   className,
   html,
   sanitizeOptions,
   testId,
+  getGlossaryEntry,
 }: ContentHtmlProps) => {
+  const [displayGlossaryEntry, setDisplayGlossaryEntry] = useState(false);
+  const [glossaryEntry, setGlossaryEntry] = useState({
+    title: 'Glossary entry not retrieved',
+    slug: '',
+    body: '',
+    link: '',
+  });
+
   const cleanHtml = useMemo(() => {
     return sanitizeHtml(html, sanitizeOptions);
   }, [html, sanitizeOptions]);
 
-  const parsedContent = htmlParser(cleanHtml, {});
+  const parsedContent = parseHtmlString(cleanHtml, {
+    replace: (node: DOMNode) => {
+      if (
+        getGlossaryEntry &&
+        node instanceof Element &&
+        node.name === 'a' &&
+        node.attribs &&
+        Object.keys(node.attribs).includes('data-glossary')
+      ) {
+        return (
+          <>
+            <a href={node.attribs.href}>{domToReact(node.children)}</a>{' '}
+            <a
+              href="#"
+              onClick={async event => {
+                event.preventDefault();
+                const glossaryEntrySlug = node.attribs.href.split('#')[1];
+                const newGlossaryEntry = await getGlossaryEntry(
+                  glossaryEntrySlug,
+                );
+                setGlossaryEntry({
+                  title: newGlossaryEntry.title,
+                  slug: newGlossaryEntry.slug,
+                  body: newGlossaryEntry.body,
+                  link: node.attribs.href,
+                });
+                setDisplayGlossaryEntry(true);
+              }}
+            >
+              ?
```

Comment:
I know I need to add some accessibility stuff for this, but wasn't sure what. So I thought I'd just let you pull me up in the PR about it.

Reply:
Fantastic, I was about to question it, and see that it's already on your radar :)

Potentially an aria-label attribute to provide a description of the link, as there's nothing else describing what it does on the page currently.

Reply:
I've added an aria-label. Wasn't sure how to phrase the label - so I'll leave this comment unresolved.

Reply:
Probably would suggest we do something like the following (I haven't included all the button props for brevity):

```tsx
<button>
  <span aria-hidden>?</span>
  <span className="govuk-visually-hidden">
    Show glossary term definition for X
  </span>
</button>
```

Where X is the actual definition.